src/test/java/Homework23LoginTests.java

```diff
@@ -0,0 +1,24 @@
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pagefactory.HomePage;
+import pagefactory.LoginPage;
+
+public class Homework23LoginTests extends BaseTest {
+
+
+    @Test (dataProvider = "IncorrectData", dataProviderClass = BaseTest.class)
+    public void loginInvalidEmailInvalidPasswordTest(String email, String incorrectPassword) throws InterruptedException {
+        provideIncorrectEmail(email);
+        provideIncorrectPassword(incorrectPassword);
+        logInButton();
+        Assert.assertEquals(driver.getCurrentUrl(), url);
+    }
+    @Test
+    public void ValidLogInCredentials() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        loginPage.login();
+        homePage.assertAvatar();
```

Comment:
Please, add Assert

**One test - One Assert**

for example 

![Clean_Test](https://github.com/EHFoshee/A45-QA-Automation-Framework/assets/104761482/775b9d62-d3f0-4d73-b638-a3fa7266fadd)

Reply:
Hi there, I am not sure why my assert is not good enough on line 21?

Reply:
HI

```
    public boolean assertAvatar() {
        WebElement usersAvatar = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img.avatar")));
        //driver.findElement(By.cssSelector("img.avatar"));
        return usersAvatar.isDisplayed();

    }}
```

However, there is no explicit assertion in your code. The line return usersAvatar.isDisplayed(); only returns a boolean value indicating whether the avatar element is displayed or not. It does not perform an actual assertion or provide any error message if the assertion fails.

Please, use an assertion to verify the visibility of the avatar, you can modify your code like this:

```
public void assertAvatar() {
    WebElement usersAvatar = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img.avatar")));
    Assert.assertTrue(usersAvatar.isDisplayed());
}
```

or

```
    @Test
    public void validLogInCredentials()  {
        LoginPage loginPage = new LoginPage(driver);
        HomePage homePage = new HomePage(driver);
        loginPage.login();
       Assert.assertTrue(homePage.assertAvatar()); 
```